### DISCUSSION TITLE
Next-Auth PassportJs Compatibility

### DISCUSSION THREAD

**Discussion by Eftychiou** [nextauthjs/next-auth#2845] [Help]
Our company started started the project using for api's the Serverless functions of NextJs, therefore we chose Next-auth for Authentication. Now that the project became much bigger, the team decided to move the Apis to an express server. The question is can we keep using Next-auth for Sign-in purposes(easy implementation with various providers) and use passportJs on Express to protect Api Routes with a JWT token ? Is this a bad approach ? Any examples of implementing it ? We are open to all sugestions.
Thanks in advance team

**Comment by balazsorban44** [comment]
Extending NextAuth.js to places outside of Next.js is planned, and there are some examples already that use it in an Express-like server. See #2294

**Reply by Eftychiou** [reply]
Do you think sending the token that was generated from nex-auth(vercel hosted) to my backend(own express serverhosted on vps) is a good approach ? And if it's okay how to we get the token in the client side and send it together with the http request ? Thanks for your early response Balazsorban44

**Reply by balazsorban44** [reply]
Check out the jwt and session callbacks. your backend probably won't need our JWT itself, but the one it creates/sends on login.

https://next-auth.js.org/configuration/callbacks

**Comment by Eftychiou** [comment]
The real problem is how to get the encrypted jwt token so it can be send  to the backend so the authentication for the api's will be done there. I tried  to get the cookie on the front-end but this can't be done because next-auth.session-token is a HttpOnly cookie . Any further ideas  ?

________________________________

**Reply by balazsorban44** [reply]
Yes, check the docs, as I suggested :wink:
https://github.com/nextauthjs/next-auth/discussions/2845#discussioncomment-1394531
